lib/collection/property-list.js

```diff
@@ -229,25 +235,29 @@ _.extend(PropertyList.prototype, /** @lends PropertyList.prototype */ {
      * Iterate on each item of this list
      */
     each: function (iterator, context) {
-        _.each(this.members, iterator, context || this.__parent);
+        _.forEach(this.members, iterator.bind(context || this));
     },
 
     /**
-     * @param {Function} rule
+     * @param {Function} iterator
      */
-    filter: function (rule, context) {
-        return _.filter(this.members, rule, context);
+    filter: function (iterator, context) {
+        return _.filter(this.members, iterator.bind(context || this));
     },
 
     /**
      * Find an item within the item group
      *
-     * @param {Function} rule
+     * @param {Function} iterator
      * @param {Object} [context]
      * @returns {Item|ItemGroup}
      */
-    find: function (rule, context) {
-        return _.find(this.members, rule, context);
+    find: function (iterator, context) {
```

Comment:
Do not introduce changes, the old code does not do a `||` for context binding, and neither does it check whether the argument is a function.